src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -167,70 +168,41 @@ private void taskScheduler() {
         LayoutView layoutView;
         CorfuRuntime corfuRuntime;
 
-        do {
-            corfuRuntime = getCorfuRuntime();
-            if (isShutdown) return;
-        } while (corfuRuntime == null);
+        corfuRuntime = getCorfuRuntime();
+        if (corfuRuntime == null) return;
 
+        corfuRuntime.invalidateLayout();
 
-        while (true) {
-            try {
-                corfuRuntime.invalidateLayout();
-
-                // Fetch the latest layout view through the runtime.
-                layoutView = corfuRuntime.getLayoutView();
-                currentLayout = layoutView.getLayout();
-                //TODO: Store this layout on disk for persistence.
-
-                // Execute the failure detection policy once.
-                failureDetectorPolicy.executePolicy(currentLayout, corfuRuntime);
-
-                // Get the server status from the policy and check for failures.
-                HashMap<String, Boolean> map = failureDetectorPolicy.getServerStatus();
-                if (!map.isEmpty()) {
-                    // Failure present. Triggering the handler.
-                    triggerHandlerDispatcher(currentLayout, map.keySet());
-                }
-
-                Thread.sleep(policyExecuteInterval);
-
-            } catch (InterruptedException err) {
-                log.debug("Sleep interrupted.");
-                if (isShutdown) {
-                    return;
-                }
-            }
-        }
+        // Fetch the latest layout view through the runtime.
+        layoutView = corfuRuntime.getLayoutView();
+        currentLayout = layoutView.getLayout();
+        //TODO: Store this layout on disk for persistence.
+
+        // Execute the failure detection policy once.
+        failureDetectorPolicy.executePolicy(currentLayout, corfuRuntime);
+
+        // Get the server status from the policy and check for failures.
+        HashMap<String, Boolean> map = failureDetectorPolicy.getServerStatus();
```

Comment:
getServerStatus gets the accumulated status of the nodes/servers since the last fetch.
So the policy keeps state of the status until another fetch occurs.